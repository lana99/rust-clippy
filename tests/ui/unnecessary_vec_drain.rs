#![allow(unused)]
#![warn(clippy::unnecessary_vec_drain)]

fn main() {
    let mut vec: Vec<i32> = Vec::new();
    //Lint
    vec.drain(..);
    vec.drain(0..vec.len());

    // Dont Lint
    let iter = vec.drain(..);
}
