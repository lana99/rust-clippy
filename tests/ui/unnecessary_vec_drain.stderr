error: unnecessary iterator `Drain` is created and dropped immedietly
  --> $DIR/unnecessary_vec_drain.rs:7:5
   |
LL |     vec.drain(..);
   |     ^^^^^^^^^^^^^ help: consider calling `clear()`: `vec.clear()`
   |
   = note: `-D clippy::unnecessary-vec-drain` implied by `-D warnings`

error: unnecessary iterator `Drain` is created and dropped immediately
  --> $DIR/unnecessary_vec_drain.rs:8:9
   |
LL |     vec.drain(0..vec.len());
   |         ^^^^^^^^^^^^^^^^^^^ help: consider calling `clear()`: `vec.clear()`

error: aborting due to 2 previous errors

